app/src/main/java/unipd/se18/ocrcamera/SearchResultActivity.java

```diff
@@ -0,0 +1,407 @@
+
+            if(ingredientsFound != null && ingredientsFound.size() > 0) {
+
+                //show ingredients list if any
+                AdapterIngredient adapter =
+                        new AdapterIngredient(getApplicationContext(), ingredientsFound);
+                mIngredientsListView.setAdapter(adapter);
+
+                //in case of results, hide dropdown suggestions
+                mAutoCompleteTextView.dismissDropDown();
+            } else {
+                //message that nothing has been found
+                mMessageTextView.setText("Nothing found");
+            }
+
+        } else {
+
+            //if the intent action is not an ACTION_SEARCH then close the activity and display an error
+            Log.e(TAG, "Activity called without ACTION_SEARCH intent");
+            Toast.makeText(getApplicationContext(), "Error doing search", Toast.LENGTH_SHORT).show();
+            finish();
+        }
+    }
+
+
+    /**
+     * Load all INCI ingredients names, set them as suggestions with a custom filter for the
+     * AutoCompleteTextView and set a suggestion click listener.
+     * The custom filter will show the suggestion in lexicographical order, privileging words that
+     * starts with the constraint query or contains a substring that starts with the constraint
+     * query.
+     */
+    private class AutoCompleteTVSetup implements Runnable {
+        @Override
+        public void run() {
+            //specify to run this code on background to avoid slowing down the UI
+            android.os.Process.setThreadPriority(Process.THREAD_PRIORITY_BACKGROUND);
+
+            //start suggesting with more than SUGGESTION_THRESHOLD chars typed
+            mAutoCompleteTextView.setThreshold(SUGGESTION_THRESHOLD);
+
+            //instance the suggestion adapter with the custom suggestion filter
+            dropDownListAdapter = new ArrayAdapter<String>(
+                    getApplicationContext(),
+                    android.R.layout.simple_list_item_1,
+                    getFullIngredientsList()){
+
+                //override getFilter to use the custom suggestions filter
+                private Filter filter;
+                @NonNull
+                @Override
+                public Filter getFilter() {
+
+                    // return filter if not null
+                    if(filter == null) {
+                        filter = new IngredientsFilter();
+                    }
+                    return filter;
+                }
+            };
+
+            mAutoCompleteTextView.setAdapter(dropDownListAdapter);
+
+
+            //on suggestion click: update ingredients listview
+            mAutoCompleteTextView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                @Override
+                public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+
```

Comment:
I'm not clear what this part does, could you explain to me what you did?

Reply:
When the user clicks a suggestion I update the text inside the AutoCompleteTextView and act like the button was clicked, causing the update of the ingredients listview

Reply:
thank you so much for the explanation.